hazelcast-jet-all/src/main/java/com/hazelcast/jet/server/JetBootstrap.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.server;
+
+import com.hazelcast.core.Cluster;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.DAG;
+import com.hazelcast.jet.Jet;
+import com.hazelcast.jet.JetInstance;
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.config.JetConfig;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.impl.util.Util;
+import com.hazelcast.jet.stream.IStreamCache;
+import com.hazelcast.jet.stream.IStreamList;
+import com.hazelcast.jet.stream.IStreamMap;
+
+import java.lang.reflect.Method;
+import java.util.function.Supplier;
+import java.util.jar.JarFile;
+
+/**
+ * A Bootstrapper for {@link JetInstance} which returns a bootstrapped Jet client instance.
+ * When used in combination with {@code jet-submit.sh}, it runs the JAR file
+ * supplied to {@code jet-submit.sh}. The wrapped instance will then add
+ * the given JAR file to all jobs submitted through that instance.
+ * <p>
+ * In the following example, the JAR file containing {@code CustomJetJob}
+ * will be submitted automatically in the {@link JetInstance#newJob(DAG)} method call as part of the
+ * {@link JobConfig} when run as {@code jet-submit.sh custom-jet-job.jar}.
+
```

Comment:
I'd also explain here, WHY we need such utility, that it's the `addClass` method, that needs to be called.